### ISSUE TITLE
davinci.ve._preferences not updated when saving updated preferences

### ISSUE THREAD

**Issue by JonFerraiolo** [maqetta/maqetta#1076]
@childsb

I discovered and fixed this on my local machine. Just logging a bug as an FYI.

I am working on another HTML editor preference and discovered that changing+saving the preference was not updating davinci.ve._preferences, which is data structure that the visual editors use to look for page editor preferences. I have now added code that updates davinci.ve._preferences whenever updated preferences are saved, but it's off in one of my local branches for another day or two before I'll merge into master and push to repository.

**Comment by childsb** [comment]
could you explain more?

Is the preference not being saved? or is it not updating the preference inside the visual editor if the visual editor is already running?

**Comment by JonFerraiolo** [comment]
preference is being saved to the server, but davinci.vi._preferences global object is not getting updated, which means visual editors that are already running (or that will run subsequently) will not get the updated preferences info because context.getPreference() returns the current value from davinci.vi._preferences.

You would think this would be trivial to fix, but I ran into a gotcha in that nearly everything in the visual editor space is instance-based (i.e., there are N visual editors, once for each open editor tab), but davinci.vi._preferences  is application global and is on the "davinci/ve" layer instead of the "davinci" layer where preferences and other application-global things are managed. I didn't want to put callbacks or listeners into any of the visual editor instances because that would mean N updates to davinci.vi._preferences if you have N editors open, and I didn't want to have the "davinci" layer have any special knowledge about the "davinci/ve" layer because that's a no-no (although we violate that rule today in various places). I concluded the cleanest way to deal with this would be to have Preferences.js look to see if the current preferences pane has a save() function and if so invoke it. This allows the HTMLEditorPreferences (which is in the VE layer) get invoked whenever the user clicks on the Save button in the preferences dialog. I'm not sure my fix is 100% solid, though. When I check in my fix, I'll @childsb within a commit note to point out the part that worries me.
